src/typecheck/arrays.ml

```diff
@@ -1,6 +1,43 @@
 
 module Id = Dolmen.Std.Id
 
+(* Ae arrays *)
+(* ************************************************************************ *)
+
+module Ae = struct
+
+  module Tff
+      (Type : Tff_intf.S)
+      (Ty : Dolmen.Intf.Ty.Ae_Array with type t := Type.Ty.t)
+      (T : Dolmen.Intf.Term.Ae_Array with type t := Type.T.t) = struct
+
+    type _ Type.err +=
+      | Bad_farray_arity : Dolmen.Std.Term.t Type.err
+
+    let parse env s =
+      match s with
+      | Type.Id { name = Simple "farray"; ns = Term } ->
+        `Ty (
+          fun ast args ->
+            Base.ty_app2 (module Type) env s Ty.array ast
```

Comment:
This code is a bit too convoluted. You should try something more along the lines of what is done in the case of the `minus` sign in the smtlib2 arithmetic (see e.g. https://github.com/Gbury/dolmen/blob/f1f05bb281826fa7c808d86fbb21c514d999edad/src/typecheck/arith.ml#L560-L568 )

Reply:
I am not sure that I understood what you meant, but I slightly changed it in the new commit.